src/Sylius/Component/Core/OrderProcessing/InventoryHandler.php

```diff
@@ -67,19 +62,13 @@ public function __construct(
      */
     public function processInventoryUnits(OrderItemInterface $item)
     {
-        $nbUnits = $item->getInventoryUnits()->count();
+        $nbUnits = $item->getUnits()->count();
 
         if ($item->getQuantity() > $nbUnits) {
             $this->createInventoryUnits($item, $item->getQuantity() - $nbUnits);
         } elseif ($item->getQuantity() < $nbUnits) {
-            foreach ($item->getInventoryUnits()->slice(0, $nbUnits - $item->getQuantity()) as $unit) {
-                $item->removeInventoryUnit($unit);
-            }
-        }
-
-        foreach ($item->getInventoryUnits() as $unit) {
-            if ($unit->getStockable() !== $item->getVariant()) {
```

Comment:
Now ``OrderItemUnit::getStockable()`` is implemented as ``return $this->orderItem->getVariant()``, so there is no need to check if ``OrderItem`` variant is the same as ``OrderItemUnit`` stockable, cause it it always true.